eks-anywhere-common/Testers/Pulumi/pulumi-tester-cronjob.yaml

```diff
@@ -0,0 +1,170 @@
+    cat << EOF > $MANIFEST_FILENAME
+    apiVersion: pulumi.com/v1
+    kind: Program
+    metadata:
+      name: eks-pulumi-operator-test
+      namespace: pulumi
+    program:
+      resources:
+        myRandomPet:
+          type: random:RandomPet
+      outputs:
+        petName: \${myRandomPet.id}
+    ---
+    apiVersion: pulumi.com/v1
+    kind: Stack
+    metadata:
+      name: eks-pulumi-operator-test
+      namespace: pulumi
+    spec:
+      stack: ${STACKPATH}
+      programRef:
+        name: eks-pulumi-operator-test
+      destroyOnFinalize: true
+      envRefs:
+        PULUMI_ACCESS_TOKEN:
+          type: Secret
+          secret:
+            name: pulumi-access-token
+            key: value
+    EOF
+
+
+    echo ""
+    echo "Deploying sample stack and program."
+    kubectl apply -f $MANIFEST_FILENAME
+
+    echo ""
+    echo "Waiting for the operator to deploy the stack."
+    sleep 10
+
+    echo ""
+    echo "Verifying that the stack exists."
+    curl \
+      --fail \
+      -H "Accept: application/vnd.pulumi+8" \
+      -H "Content-Type: application/json" \
+      -H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+      https://api.pulumi.com/api/stacks/${STACKPATH}
+
+    echo ""
+    echo "Destroying K8s Stack resource"
+    kubectl delete -n pulumi stacks/eks-pulumi-operator-test
+
+    echo ""
+    echo "Waiting for the operator to remove the stack"
+    sleep 10
+
+    echo ""
+    echo "Verifying the stack no longer exists"
+    STATUSCODE=$(curl \
+      -s \
+      -o /dev/null \
+      --w "%{http_code}" \
+      -H "Accept: application/vnd.pulumi+8" \
+      -H "Content-Type: application/json" \
+      -H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+      https://api.pulumi.com/api/stacks/${STACKPATH}
+    )
+
+    if test $STATUSCODE -ne 404; then
+      echo "ERROR: Expected HTTP status code 404 from the Pulumi Cloud API when querying the stack. Got HTTP status code $STATUSCODE instead."
+      false
+    fi
+
+    # This is for purely for running the script locally. Since the K8s tester Job is
+    # run in an ephemeral container, deleting the file is unnecessary in that
+    # context:
+    echo ""
+    echo "Deleting test manifest file '${MANIFEST_FILENAME}'"
+    rm ${MANIFEST_FILENAME}
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: pulumi-k8s-operator-test
+  namespace: pulumi
+spec:
+  schedule: "9 9 * * *"
+  jobTemplate:
+    spec:
+      activeDeadlineSeconds: 900
+      template:
+        spec:
+          serviceAccountName: pulumi-tester
+          containers:
+            - command:
+                - bash
+                - /scripts/pulumi-k8s-operator-test.sh
+              image: "bitnami/minideb:bookworm"
+              name: script
+              env:
+                - name: PULUMI_ACCESS_TOKEN
+                  valueFrom:
+                    secretKeyRef:
+                      name: pulumi-access-token
+                      key: value
```

Comment:
@jkodroff This doesnt seem to match with your `ExternalSecret` k8s rsource. If the `key` is `pulumi-access-token`, it should match.

Reply:
I believe this is now fixed.

Reply:
I would recommend giving a meaningful name like `accessToken` for key.